### PR TITLE
[superseded] fix #15628

### LINKED ISSUE #15628: coro lib doesn't work with orc/arc
Coroutines doesn't compile with gc orc/arc

coro.nim uses nimGC_setStackBottom and references GC_addStack, etc. implemented in gc_common.nim which is not included by orc / arc

as of 
Nim Compiler Version 1.5.1 [Windows: amd64]
git hash: 091ec916fd2737d495c5ef54afb5964ee62c049d

### PR DESCRIPTION
**Pull request by ringabout**
Thank @Clyybber for hints!

### CODE REVIEW

**Comment by ringabout** [comment]
freebsd failed

```
/home/build/Nim/nimcache/tests/coroutines/tgc.nim_0d61f8370cad1d412f80b84d143e1257/stdlib_coro.nim.c:393:22: error: array type 'jmp_buf' (aka 'long [11]') is not assignable
        (*dest).execContext = src.execContext;
        ~~~~~~~~~~~~~~~~~~~ ^
CC: tgc.nim
1 error generated.
Error: execution of an external compiler program 'clang -c  -w -ferror-limit=3 -U_FORTIFY_SOURCE
```

File: lib/pure/coro.nim [hunk 1]
```diff
@@ -32,6 +32,10 @@ import lists
 include system/timers
 
 const defaultStackSize = 512 * 1024
+const useOrcArc = defined(gcOrc) or defined(gcArc)
```
Comment (Araq) [inline lib/pure/coro.nim]: ``gcArc`` is defined for ``--gc:orc`` too.

Comment (ringabout) [inline lib/pure/coro.nim]:
```suggestion
const useOrcArc = defined(gcArc)
```

File: lib/system/arc.nim [hunk 2]
```diff
@@ -226,6 +226,7 @@ template tearDownForeignThreadGc* =
   ## With ``--gc:arc`` a nop.
   discard
 
+
```
Comment (ringabout) [inline lib/system/arc.nim]:
```suggestion
```

File: lib/pure/coro.nim [hunk 3]
```diff
@@ -32,6 +32,10 @@ import lists
 include system/timers
 
 const defaultStackSize = 512 * 1024
+const useOrcArc = defined(gcArc)
```
Comment (ringabout) [inline lib/pure/coro.nim]:
```suggestion
const useOrcArc = defined(gcArc) or defined(gcOrc)
```

File: lib/pure/coro.nim (hunk 1 above)
Comment (ringabout) [reply lib/pure/coro.nim]: `nimGC_setStackBottom` is defined in `mm.nim` already, so I can't define them in `arc.nim`. I have to define them in `coro.nim`. So I need `gcOrc` .

**Pull request closed without merging**